src/api/entitlements.js

```diff
@@ -356,30 +362,45 @@ export class Entitlement {
 
   /**
    * @param {?Object} json
+   * @param {!../utils/jwt.JwtHelper} jwtHelper
    * @return {!Entitlement}
    */
-  static parseFromJson(json) {
+  static parseFromJson(json, jwtHelper) {
     if (!json) {
       json = {};
     }
     const source = json['source'] || '';
     const products = json['products'] || [];
     const subscriptionToken = json['subscriptionToken'];
-    return new Entitlement(source, products, subscriptionToken);
+    let subscriptionTokenContents;
+    try {
+      subscriptionTokenContents = subscriptionToken
+        ? jwtHelper.decode(subscriptionToken)
+        : null;
+    } catch (e) {
+      subscriptionTokenContents = null;
```

Comment:
Maybe set to the passed in contents instead?

Reply:
So in these cases, the JWT decode has failed (381), or, there were was nothing to decode (379). 

I'm sorry to say this, but I don't understand what you're suggesting - what are the passed in contents in the latter case? Do you mean, in the former case, put the non-decodable string contents into the field intended to hold decoded token contents? That doesn't seem right.

Reply:
oh, nvm. the token itself is still accessible via `subscriptionToken`; so yeah, this doesnt make sense. ignore